### PR TITLE
Show error span at the last non-whitespace character of the file if the error is at EOF.

### PR DESCRIPTION
**Pull request by hoanhtien**
This fixes https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/44.

### CODE REVIEW

File: typescript/libs/view_helpers.py [hunk 1]
```diff
@@ -315,4 +315,20 @@ def change_count(view):
...
             return info.modify_count
         else:
-            return view.change_count()
\ No newline at end of file
+            return view.change_count()
+
+def last_non_whitespace_position(view):
+    """
+    Returns the position of the last non-whitespace character of <view>.
+    Returns -1 if <view> contains no non-whitespace character.
```
Comment (DanielRosenwasser) [inline typescript/libs/view_helpers.py]: only contains whitespace characters

File: typescript/libs/view_helpers.py [hunk 2]
```diff
@@ -315,4 +315,20 @@ def change_count(view):
...
+    """
+    pos = view.size() - 1
+    while pos >= 0 and view.substr(pos).isspace():
+        pos -= 1
+    return pos
+
+def last_character_region(view):
+    """Returns the last non whitespace character under <sublime.Region> format"""
```
Comment (DanielRosenwasser) [inline typescript/libs/view_helpers.py]: Returns a <sublime.Region> for the last non-whitespace character

File: typescript/listeners/idle.py [hunk 3]
```diff
@@ -89,16 +89,22 @@ def show_errors(self, diagno_event_body, syntactic):
                         start = view.text_point(line, offset)
                         end = view.text_point(end_line, end_offset)
                         if end <= view.size():
-                            region = sublime.Region(start, end)
+                            if start == view.size() and start == end:
```
Comment (DanielRosenwasser) [inline typescript/listeners/idle.py]: Leave a comment explaining why you need to do this

File: typescript/libs/view_helpers.py [hunk 4]
```diff
@@ -315,4 +315,20 @@ def change_count(view):
...
\ No newline at end of file
+            return view.change_count()
+
+def last_non_whitespace_position(view):
+    """
+    Returns the position of the last non-whitespace character of <view>.
+    Returns -1 if <view> only contains non-whitespace characters.
+    """
+    pos = view.size() - 1
+    while pos >= 0 and view.substr(pos).isspace():
```
Comment (zhengbli) [inline typescript/libs/view_helpers.py]: ``is_space()``

**Comment by zhengbli** [comment]
:+1:

**Pull request merged**